### PR TITLE
feat(groups): break down groups/all page in smaller views

### PR DESCRIPTION
**Pull request by hypeJunction**
Make it easier to override individual parts and add new listing types
to the groups/all page

### CODE REVIEW

File: mod/groups/views/default/groups/listing/discussion.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
```
Comment (jeabakker) [inline mod/groups/views/default/groups/listing/discussion.php]: why have this wrapper view in groups and not directly in discussions?

**Comment by jeabakker** [comment]
I love the idea

File: mod/groups/views/default/resources/groups/all.php [hunk 2]
```diff
@@ -9,66 +9,12 @@
...
-			'full_view' => false,
-			'no_results' => elgg_echo('groups:none'),
-			'distinct' => false,
-		));
-		break;
+if (!elgg_view_exists("groups/listing/$selected_tab")) {
+	$selected_tab = 'newest';
 }
 
+$content = elgg_view("groups/listing/$selecte_tab", $vars);
```
Comment (jdalsem) [inline mod/groups/views/default/resources/groups/all.php]: typo in selected tab

File: mod/groups/views/default/groups/listing/discussion.php (hunk 1 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/listing/discussion.php]: Saves adding exceptions to the resource view

File: mod/groups/views/default/resources/groups/all.php (hunk 2 above)
Comment (hypeJunction) [inline mod/groups/views/default/resources/groups/all.php]: Eh. Fixed locally. Didn't push I guess

File: mod/groups/views/default/groups/listing/discussion.php (hunk 1 above)
Comment (jeabakker) [inline mod/groups/views/default/groups/listing/discussion.php]: but in the resource view you check if the selected tab view exists. What exceptions are you revering to?

**Review by juho-jaakkola — APPROVED** [review]

**Review by jdalsem — CHANGES_REQUESTED** [review]
just the selected tab typo... then it is OK by me

File: mod/groups/views/default/groups/listing/discussion.php (hunk 1 above)
Comment (hypeJunction) [inline mod/groups/views/default/groups/listing/discussion.php]: It looks for `groups/listing/$selected_tab`

**Comment by hypeJunction** [comment]
Typo is fixed

File: mod/groups/views/default/groups/listing/discussion.php (hunk 1 above)
Comment (jeabakker) [inline mod/groups/views/default/groups/listing/discussion.php]: excactly so this file can be in de discussions plugin, yes/no?

Comment (hypeJunction) [inline mod/groups/views/default/groups/listing/discussion.php]: I see your point. I am not sure. I don't think I personally care either way.

Comment (hypeJunction) [inline mod/groups/views/default/groups/listing/discussion.php]: I suppose we will be moving the tab registration over to the discussion plugin eventually, and we have this view in the groups plugin, we will be stuck with it. I will move to discussions.

Comment (jeabakker) [inline mod/groups/views/default/groups/listing/discussion.php]: thx ;)

Comment (hypeJunction) [inline mod/groups/views/default/groups/listing/discussion.php]: Done

**Review by mrclay — APPROVED** [review]

**Pull request merged**